### PR TITLE
Add Python 3.6, drop Python 3.3

### PR DESCRIPTION
**Pull request by jamadden**
Three commits:

1) Standard project gardening.
2) Reach 100% coverage. This involved quite a few new tests for the state machine.
3) Clean up all the whitespace violations in `__init__.py` (of which there were a lot). This revealed some more coverage holes so it needed some new tests.

There's one test that gives different results on Py2 vs Py3. It's a pure function, so it has to be an overflow or division difference; I haven't looked into it any more closely than that because I'm not sure what the right answer should actually be without research. But perhaps this module would be a candidate for `from __future__ import division`.

### CODE REVIEW

**Comment by jamadden** [comment]
> But perhaps this module would be a candidate for `from __future__ import division`.

That definitely makes the difference. If I do that to Python 2, I get the same answer as Python 3; no other tests break.

File: src/zope/datetime/__init__.py [hunk 1]
```diff
@@ -17,281 +17,297 @@
...
+    ts = _time.time() if ts is None else ts
+    year, month, day, hh, mm, ss, wd, _y, _z = _time.gmtime(ts)
+    return "%s, %02d %3s %4d %02d:%02d:%02d GMT" % (
+        weekday_abbr[wd],
+        day, monthname[month],
+        year,
+        hh, mm, ss)
+
+class DateTimeError(Exception):
+    "Date-time error"
```
Comment (icemac) [inline src/zope/datetime/__init__.py]: PEP-8 thinks there should be two newlines between the class definitions.
PEP-257 wants to have triple quotes.

Comment (jamadden) [inline src/zope/datetime/__init__.py]: I tackled those things (for sure the quotes) in #5.

**Comment by jamadden** [comment]
> But I wonder who uses this package.

Among zopefoundation repos, I [find actual uses](https://github.com/search?q=org%3Azopefoundation+%22zope.datetime%22&type=Code) in `zope.formlib`, `zope.app.file`, `zope.dublincore`, and `zope.app.publisher`. Some use it for parsing, some use it for formatting.

> It seems to re-invent pytz and some other packages. 

Or predate most such packages? This was first released standalone in 2007, but existed before then. pytz goes back to 2009, it looks like, and iso8601 also looks like it goes back to sometime in 2007.

> Or did I miss some functions which are unique here?

The pre-defined formats are nice. I think the parsing is *extremely* lenient compared to some other parsers (which may or may not always be an advantage).

**Comment by jamadden** [comment]
Thanks! I've rebased #5 on master now that this is merged.

**Comment by icemac** [comment]
> Or predate most such packages?

Thank you for the explanation, I was not aware that the other packages are so young. To my mind they seem to have been there since ever. :)

**Pull request merged**